src/cmd/flux-job-validator.py

```diff
@@ -0,0 +1,109 @@
+###############################################################
+# Copyright 2021 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+import os
+import sys
+import logging
+import json
+import argparse
+import flux
+from flux.job.validator import JobValidator
+
+
+LOGGER = logging.getLogger("flux-job-validator")
+
+
+class HelpAction(argparse.Action):
+    """
+    Copy of argparse._HelpAction, so that `--help` can be added
+    after initial parse_known_args() in JobValidator constructor
+    """
+
+    def __init__(
+        self,
+        option_strings,
+        dest=argparse.SUPPRESS,
+        default=argparse.SUPPRESS,
+        xhelp=None,
+    ):
+        super(HelpAction, self).__init__(
+            option_strings=option_strings,
+            dest=dest,
+            default=default,
+            nargs=0,
+            help=xhelp,
+        )
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        parser.print_help()
+        parser.exit()
+
+
+@flux.util.CLIMain(LOGGER)
+def main():
+
+    parser = argparse.ArgumentParser(
+        prog="flux-job-validator",
+        formatter_class=flux.util.help_formatter(),
+        description="Validate Flux jobs from lines of JSON input on stdin",
+        add_help=False,
+    )
+    script_group = parser.add_argument_group("Program options")
+    script_group.add_argument(
+        "--jobspec-only",
```

Comment:
Neat!  Really neat.

I wonder if it would be interesting to users to be able to play with feasibility without actually submitting anything?   Maybe we could add a `--feasibility` option like `--dry-run` to `flux mini submit` that does the equivalent of
```
$ flux mini submit --dry-run hostname | flux job-validator --jobspec-only --plugins feasibility
{"errnum": 0}
```
and parses the returned JSON?   Just a thought (but probably not for this PR)

Reply:
That's a good idea! The validator is actually a python class now so it could embedded directly in tools like `flux-mini.py` (which was part of the reason for doing it that way)